src/PolyhedralGeometry/Polyhedron/standard_constructions.jl

```diff
@@ -1218,18 +1190,1083 @@ julia> nvertices(rand_subpolytope(cube(3), 5))
 ```
 """
 function rand_subpolytope(P::Polyhedron{T}, n::Int; seed=nothing) where T<:scalar_types
-  if !is_bounded(P)
-    throw(ArgumentError("rand_subpolytope: Polyhedron unbounded"))
-  end
+  @req is_bounded(P) "Polyhedron is unbounded, has to be bounded"
   nv = nvertices(P)
-  if n>nv
-    throw(ArgumentError("rand_subpolytope: number of vertices requested too high"))
-  end
+  @req n <= nv "Number of vertices requested too high"
   opts = Dict{Symbol,Any}()
-  if seed != nothing
+  if !isnothing(seed)
     opts[:seed] = convert(Int64, seed)
   end
   pm_matrix = Polymake.polytope.rand_vert(P.pm_polytope.VERTICES, n; opts...)
   pm_obj = Polymake.polytope.Polytope{_scalar_type_to_polymake(T)}(VERTICES=pm_matrix)::Polymake.BigObject
   return Polyhedron{T}(pm_obj, coefficient_field(P))
 end
+
+@doc raw"""
+    SIM_body_polytope(alpha::AbstractVector)
+
+Produce an $n$-dimensional SIM-body as generalized permutahedron in 
+$(n+1)$-space. SIM-bodies are defined in [GK14](@cite), but the input needs to 
+be descending instead of ascending, as used in [JKS22](@cite), i.e. `alpha` has parameters
+$(a_1,\dots,a_n)$ such that $a_1 \geq \dots \geq a_n \geq 0$.  
+
+# Example
+To produce a $2$-dimensional SIM-body, use for example the following code. 
+Note that the polytope lives in $3$-space, so we project it down to $2$-space 
+by eliminating the last coordinate. 
+
+```jldoctest
+julia> s = SIM_body_polytope([3,1])
+Polyhedron in ambient dimension 3
+
+julia> p = convex_hull(map(x->x[1:dim(s)],vertices(s)))
+Polyhedron in ambient dimension 2
+
+julia> vertices(p) 
+5-element SubObjectIterator{PointVector{QQFieldElem}}:
+ [0, 0]
+ [3, 0]
+ [3, 1]
+ [0, 3]
+ [1, 3]
+```
+"""
+function SIM_body_polytope(alpha::Vector) 
+    n = length(alpha)
+    @req n >= 1 "dimension must be at least 1"
+    @req issorted(-alpha) "input is not descending"
```

Comment:
```suggestion
    @req issorted(alpha; rev=true) "input is not descending"
```
Doesn't allocate a new vector